src/network/protocol.rs

```diff
@@ -330,6 +329,50 @@ pub struct FixedIp {
     pub subnet_id: String,
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Default, Ord, PartialOrd, Hash)]
+pub struct MacAddress(macaddr::MacAddr6);
+
+impl MacAddress {
+    pub fn is_nil(&self) -> bool {
```

Comment:
I'm curious why you need this if you have Deref

Reply:
It seems to be needed as the serde line:

```rust
...
 #[serde(skip_serializing_if = "MacAddress::is_nil")]
pub mac_address: MacAddress,
...
```

specifies a path to a function, so if I remove the `fn is_nil`, then it gives the error:

```
error[E0599]: no function or associated item named `is_nil` found for struct `MacAddress` in the current scope
   --> src/network/protocol.rs:461:35
    |
333 | pub struct MacAddress(macaddr::MacAddr6);
    | --------------------- function or associated item `is_nil` not found for this struct
...
461 |     #[serde(skip_serializing_if = "MacAddress::is_nil")]
    |                                   ^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `MacAddress`
```

The alternative seems to be, annotating the serde field with:

```rust
...
 #[serde(skip_serializing_if = "macaddr::MacAddr6::is_nil")]
pub mac_address: MacAddress,
...
```

but I thought avoiding having to leak the underlying type directly was worth it.